python/cudf/cudf/tests/test_copying.py

```diff
@@ -52,3 +53,324 @@ def test_null_copy():
     col = Series(np.arange(2049))
     col[:] = None
     assert len(col) == 2049
+
+
+# TODO: Make use of set_option context manager
+# once https://github.com/rapidsai/cudf/issues/12736
+# is resolved.
+
+
+def test_series_setitem_cow_on():
```

Comment:
This function tests multiple different cases. Can it be split up so that they are tested independently?

Reply:
Done